File: Neetcode150/Stack/ValidParentheses/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        check = { ")" : "(", "]" : "[", "}" : "{" }
        stack = []

        for p in s:
            if p not in check:
                stack.append(p)
                continue
            if not stack or stack[-1] != check[p]:
                return False
            
            stack.pop()
        
        return len(stack) == 0

File: Neetcode150/Stack/ValidParentheses/ValidParentheses.cs
public class Solution {
    public bool IsValid(string s) {
        var stack = new Stack();
        var check = new Dictionary<char, char>
        {
            { '}', '{' },
            { ')', '(' },
            { ']', '[' }
        };


        foreach (char p in s) {
            if (!check.ContainsKey(p)) {
                stack.Push(p);
                continue;
            } 
            while (stack.Count == 0 || (char)stack.Peek() != check[p]) {
                return false;
            }
            stack.Pop();
        }

        return stack.Count == 0;
    }
